src/sys/pthread.rs

```diff
@@ -36,3 +38,22 @@ pub fn pthread_kill<T: Into<Option<Signal>>>(thread: Pthread, signal: T) -> Resu
     let res = unsafe { libc::pthread_kill(thread, sig) };
     Errno::result(res).map(drop)
 }
+
+/// Obtain the name of the thread
+///
+/// On linux the name cannot exceed 16 length including null terminator
+#[cfg(all(target_os = "linux", not(target_env = "musl")))]
+pub fn pthread_getname_np(thread: Pthread) -> Result<String> {
+    let mut name = [0u8; 16];
```

Comment:
The 16 bytes thing is Linux specific.  On NetBSD it's controlled by the `PTHREAD_MAX_NAMELEN_NP` constant, currently 32 bytes.  On FreeBSD it's not clearly documented, but the limit is `MAXCOMLEN + 1`, which is 20.

Reply:
very true but as mentionned in the commit message I focus on Linux for now.